pkg/destroy/aws/iamhelpers.go

```diff
@@ -249,8 +249,19 @@ func deleteIAMInstanceProfile(ctx context.Context, client *iam.IAM, profileARN a
 	}
 
 	logger = logger.WithField("arn", profileARN.String())
```

Comment:
Right before deleting the Instance profile by name, this code block above tries to dissociate a Role from the Instance profile. Double checking here to see if that operation is allowed when "shared" tag is present on the Instance profile (based on my understanding, it should not be an issue.)

Reply:
This is what I found when I looked for information to understand the association between roles and instance profiles:
`An instance profile can contain only one IAM role, although a role can be included in multiple instance profiles. This limit of one role per instance profile cannot be increased. You can remove the existing role and then add a different role to an instance profile.`

So, after dissociating a role, it is possible that the instance profile is not deleted due to the shared tag.

Reply:
This is a bit of new territory on the instance profile being tag as shared (or tagged at all). But disassociating the resources should be fine. The only task we are accomplishing with tagging the instance profiles is ensuring that they remain or get cleaned up properly.